listener.py

```diff
@@ -28,41 +30,35 @@
 # lag to make sure the blocks we see are settled
 BLOCK_NUMBER_LAG = 5
 
+primary_rpc = os.getenv("RPC_URL")
+secondary_rpc = os.getenv("SECONDARY_RPC_URL")
+
+if os.getenv("RPC_URL") is None:
+    raise RuntimeError("Missing primary RPC environment variable: RPC_URL. ")
```

Comment:
this is weird. You do the getenv twice. Maybe better to move it to a function get_rpc_url that is lazy and would fail the first time.
It's also ugly to have these actions run when the module is loaded.